src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-perf/csvjsonconvertor.py

```diff
@@ -11,6 +11,42 @@
 
 log = helix.logs.get_logger()
 
+# read a row of csv data into dict
+def add_row(test, value, csvdict):
+    """
+        test names are of the form
+        System.ComponentModel.Tests.Perf_TypeDescriptorTests.GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))
+        where the string before the first '(' are the identifiers for the test and the string after is function metadata
+        strip off the metadata when generating the namespace/function identifiers and combine them when again for function identifier
+        eg.: System.ComponentModel.Tests.Perf_TypeDescriptorTests.GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))
+        will be converted to following identifiers
+        [System, ComponentModel, Tests, Perf_TypeDescriptorTests, GetConverter(typeToConvert: typeof(bool), expectedConverter: typeof(System.ComponentModel.BooleanConverter))]
+    """
+
+    funcMeta = test.split('(')
+    funcMeta = funcMeta[1:]
+    test = test.split('(')[0]
+    identifiers = test.split('.')
+    funcName = identifiers[len(identifiers) - 1]
+    for data in funcMeta:
+        funcName = funcName + '(' + data
+
+    identifiers[len(identifiers) - 1] = funcName
+    i=0
+    currdict = csvdict
+    while (i<len(identifiers) -1):
+        if identifiers[i] not in currdict:
+            currdict[identifiers[i]] = dict()
+
+        currdict = currdict[identifiers[i]]
+        i=i+1
+
+    if identifiers[i] not in currdict:
```

Comment:
I again feel like this isn't working as expected... Matter of fact, it's going to crash weirdly. When the loop ends, i == len(idendfiers)... idenfiers[len(identifiers)] is going to return "None", which you then weirdly create a list for... So "pizza.pepperoni" become cvsdict = { "pizza": { "peperoni" : { null : [value]}}, which is very strange to have a null keyed dictionary.